src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,269 @@
+package duke.util;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
+import duke.Duke;
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+
+/**
+ * The class that parses user input and returns parsed objects.
+ */
+public class Parser {
+
+    /**
+     * The method takes a user input and parse out all necessary parts needed
+     * to execute user command and finally returns a Command object that can be executed
+     * @param input A line of user input
+     * @return A Command object which can then be called to execute user command
+     * @throws DukeException When the input is empty String or the command is invalid
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        String[] splited = input.strip().split(" +");
+        if(splited.length < 1) {
+            throw new DukeException("The input cannot be empty!");
+        }
+
+        String cmdType = splited[0].toLowerCase();
+        switch (cmdType) {
+        case "bye":
+            return new CommandBye();
+        case "done":
+            return new CommandDone(parseIndex(splited));
+        case "delete":
+            return new CommandDelete(parseIndex(splited));
+        case "list":
+            return new CommandList();
+        case "todo": {
+            String description = parseDescription(splited);
+            return new CommandAdd(new Task(description));
+        }
+        case "deadline": {
+            String description = parseDescription(splited);
+            LocalDateTime dateTime = parseDeadlineDateTime(splited);
+            return new CommandAdd(new Deadline(description, dateTime));
+        }
+        case "event": {
+            String description = parseDescription(splited);
+            LocalDateTime[] dateTimes = parseEventDateTime(splited);
+            return new CommandAdd(new Event(description, dateTimes[0], dateTimes[1]));
+        }
+        case "find": {
+            String keyword = parseKeyword(splited);
+            return new CommandFind(keyword);
+        }
+        default:
+            throw new DukeException("The command is invalid");
+        }
+
+    }
+
+    /**
+     * Parse and return the task index of CommandDelete and CommandDone.
+     * @param strArr String array that contains splited input string.
+     * @return parsedIndex
+     * @throws DukeException when the index is missing from the input.
+     */
+    public static int parseIndex(String[] strArr) throws DukeException {
+        if(strArr.length < 2) {
```

Comment:
According to the coding standard, Java reserved words should be followed by a white space, eg. `if (condition) {` and `for (i = 0; i < 10; i++) {` I noticed the same issue in several other places too.